launcher/net/HttpMetaCache.cpp

```diff
@@ -121,6 +126,15 @@ auto HttpMetaCache::resolveEntry(QString base, QString resource_path, QString ex
         SaveEventually();
     }
 
+    // Get rid of old entries, to prevent cache problems
+    auto current_time = QDateTime::currentMSecsSinceEpoch();
+    auto remote_time = QDateTime::fromString(entry->remote_changed_timestamp).toMSecsSinceEpoch();
```

Comment:
Wouldn't this invalidate icons even if they were downloaded moments ago, just because the server reported a `Last-Modified` in the past?

Reply:
See [Sodium on Modrinth](https://cdn-raw.modrinth.com/data/AANobbMI/icon.png) as an example. `Last-Modified` is `Sun, 03 Jan 2021 00:53:33 GMT`.

I would propose the following behavior: If the remote server provides a `Cache-Control` header, or even better an `Expires` header, the launcher should store those and use them to decide when something should get invalidated.
If the remote does not provide those headers, then just default to 1 week as it is rn.

Reply:
ehh, i tried implementing that, sorry if i somehow failed, i never touched any of this stuff before >//<

I also raised the priority of `Cache-Control` in relation to `Expires`, because if the former is present, [it apparently ignores the latter](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expires)